common/db_test.go

```diff
@@ -578,3 +582,311 @@ func TestDBUnbondingLocks(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	sessionID, ticket, sig, recipientRand := defaultWinningTicket(t)
+	err = dbh.StoreWinningTicket(sessionID, ticket, sig, recipientRand)
+	if err != nil {
+		t.Fatalf("unexpected errro storing ticket: %v", err)
+	}
+
+	tickets, sigs, recipientRands, err := dbh.LoadWinningTickets(sessionID)
+	if err != nil {
+		t.Fatalf("unexpected error loading tickets: %v", err)
+	}
+	if len(tickets) != 1 || len(sigs) != 1 || len(recipientRands) != 1 {
+		t.Errorf("expected one item per slice but got unexpected number of results. tickets: %d, sigs: %d, recipientRands: %d", len(tickets), len(sigs), len(recipientRands))
+	}
+	actualTicket := tickets[0]
+	actualSig := sigs[0]
+	actualRecipientRand := recipientRands[0]
+
+	assert := assert.New(t)
+	assert.Equal(ticket, actualTicket)
+	assert.Equal(sig, actualSig)
+	assert.Equal(recipientRand, actualRecipientRand)
+}
+
+func TestLoadtWinningTicket_GivenStoredTicketsFromDifferentSessions_OnlyLoadsFromSpecificSessionID(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	// Two tickets in the first session
+	firstSessionID := "first session"
+	_, ticket0, sig0, recipientRand0 := defaultWinningTicket(t)
+	err = dbh.StoreWinningTicket(firstSessionID, ticket0, sig0, recipientRand0)
+	if err != nil {
+		t.Fatalf("unexpected errro storing ticket: %v", err)
+	}
+
+	_, ticket1, sig1, recipientRand1 := defaultWinningTicket(t)
+	err = dbh.StoreWinningTicket(firstSessionID, ticket1, sig1, recipientRand1)
+	if err != nil {
+		t.Fatalf("unexpected errro storing ticket: %v", err)
+	}
+
+	// One ticket in the second session
+	secondSessionID := "second session"
+	_, ticket2, sig2, recipientRand2 := defaultWinningTicket(t)
+	err = dbh.StoreWinningTicket(secondSessionID, ticket2, sig2, recipientRand2)
+	if err != nil {
+		t.Fatalf("unexpected errro storing ticket: %v", err)
+	}
+
+	tickets, sigs, recipientRands, err := dbh.LoadWinningTickets(firstSessionID)
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Len(tickets, 2)
+	assert.Len(sigs, 2)
+	assert.Len(recipientRands, 2)
+	assert.Equal(ticket0, tickets[0])
+	assert.Equal(sig0, sigs[0])
+	assert.Equal(recipientRand0, recipientRands[0])
+	assert.Equal(ticket1, tickets[1])
+	assert.Equal(sig1, sigs[1])
+	assert.Equal(recipientRand1, recipientRands[1])
+}
+
+func TestLoadtWinningTicket_GivenNonexistentSessionID_ReturnsEmptySlicesNoError(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	tickets, sigs, recipientRands, err := dbh.LoadWinningTickets("some sessionID")
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Len(tickets, 0)
+	assert.Len(sigs, 0)
+	assert.Len(recipientRands, 0)
+}
+
+func TestLoadtWinningTicket_GivenEmptySessionID_ReturnsEmptySlicesNoError(t *testing.T) {
```

Comment:
If we ignore that fact that a `pm` recipient will use `recipientRandHash` as the sessionID, technically the DB could accept an empty string as the sessionID and return non-empty slices right?

Reply:
You're right. In this case I was aiming at making sure that we just get an empty result set rather than an error. I didn't find it interesting to add a test where there would be tickets stored with `sessionID = ""`, since I believe the combination of the tests we already have covers it. But please let me know if you think we need this additional test case and I'll add it :)

Reply:
👍 I don't think an additional test case is necessary - I just wanted to clarify that an empty sessionID should return an empty slice in this test case, but does not necessarily always return an empty slice